presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -269,11 +281,9 @@ private static Partitioner createPartitioner(PartitioningHandle partitioning, in
             PlanFragment fragment,
             PrestoSparkTaskExecutorFactoryProvider executorFactoryProvider,
             CollectionAccumulator<SerializedTaskStats> taskStatsCollector,
-            TableWriteInfo tableWriteInfo)
+            TableWriteInfo tableWriteInfo,
+            Map<PlanFragmentId, Broadcast<List<PrestoSparkSerializedPage>>> broadcastInputs)
     {
-        // TODO: Possible in case of a broadcast join
-        checkArgument(fragment.getRemoteSourceNodes().isEmpty(), "source task with remote sources is not supported");
```

Comment:
nit: is it possible to check all remote sources node is reading from broadcast source? If the check is not easy then no worry.

Reply:
Refactored inputs check in a separate method. Calling it here as well.